EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/attributeInput/AttributeInput.java

```diff
@@ -0,0 +1,282 @@
+	private final TypeTreeNode attribute;
+	private String input = "";
+	private static Random random = new Random();
+
+	final List<ProbabilityDistributionEnum> availableMethods = new ArrayList<>();
+
+	private ProbabilityDistributionEnum selectedMethod;
+
+	static final Logger logger = Logger.getLogger(AttributeInput.class);
+
+	AttributeInput(TypeTreeNode attribute) {
+		this.attribute = attribute;
+	}
+
+	/**
+	 * Returns an AttributeInput-Object fitting to the attribute type.
+	 *
+	 * @param attribute the AttributeInput should be chosen for
+	 * @return a subclass of AttributeInput
+	 */
+	public static AttributeInput attributeInputFactory(TypeTreeNode attribute) {
+		switch (attribute.getType()) {
+			case STRING:
+				return new StringAttributeInput(attribute);
+			case INTEGER:
+				return new IntegerAttributeInput(attribute);
+			case FLOAT:
+				return new FloatAttributeInput(attribute);
+			case DATE:
+				return new DateAttributeInput(attribute);
+			default:
+				attribute.setType(AttributeTypeEnum.STRING);
+				return new StringAttributeInput(attribute);
+		}
+	}
+
+	/**
+	 * Getter for the attribute associated to the AttributeInput-Instance.
+	 *
+	 * @return an attribute
+	 */
+	public TypeTreeNode getAttribute() {
+		return attribute;
+	}
+
+	/**
+	 * Get the type of the attribute associated to the AttributeInput-Instance.
+	 *
+	 * @return the type of the attribute
+	 */
+	public AttributeTypeEnum getAttributeType() {
+		return this.getAttribute().getType();
+	}
+
+	/**
+	 * Get the name of the attribute associated to the AttributeInput-Instance.
+	 *
+	 * @return a String containing the name
+	 */
+	public String getAttributeName() {
+		return this.getAttribute().getName();
+	}
+
+	/**
+	 * Returns a list containing all (probability) methods supported by the instance.
+	 * This differs from e.g. StringAttributeInput to IntegerAttributeInput.
+	 *
+	 * @return a list with all supported methods (from ProbabilityDistributionEnum)
+	 */
+	public List<ProbabilityDistributionEnum> getAvailableMethods() {
+		return availableMethods;
+	}
+
+	/**
+	 * Returns the currently selected method for the computation of a value.
+	 *
+	 * @return the by the user selected method
+	 */
+	public ProbabilityDistributionEnum getSelectedMethod() {
+		return selectedMethod;
+	}
+
+	/**
+	 * Setter for the currently selected method.
+	 *
+	 * @param selectedMethod that should be set
+	 */
+	public void setSelectedMethod(ProbabilityDistributionEnum selectedMethod) {
+		this.selectedMethod = selectedMethod;
+	}
+
+	/**
+	 * Getter for the current input.
+	 *
+	 * @return the current input
+	 */
+	public String getInput() {
+		return input;
+	}
+
+	/**
+	 * Returns the current input, in case this is empty, a default value defined by the class is returned.
+	 *
+	 * @return a string containing the current input or the default input
+	 */
+	String getInputOrDefault() {
+		if(input != null && !input.isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') 'if' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AWhitespace_Around_BP2016)